### PR TITLE
Syzygy code style massage

### PR DESCRIPTION
**Pull request by lucasart**
No functional change.

### CODE REVIEW

**Comment by lucasart** [comment]
Verified this with syzygy 5-men, as follows:
```
$ ./stockfish
Stockfish 100416 64 BMI2 by T. Romstad, M. Costalba, J. Kiiski, G. Linscott
$ uci
...
uciok
$ setoption name SyzygyPath value /home/lucas/Chess/syzygy
info string Found 145 tablebases.
$ bench
...
Nodes searched  : 7378286
```
Same node count for both versions.

**Comment by zamar** [comment]
I have mixed feelings about this type of changes. I'm not saying absolutely no to this patch, but...

So far in my mind "syzygy bases" have just been a kind of 3rd party library that we use as an oracle and that seems to work perfectly. We've fixed a few very minor bugs, but that's really it...

When it comes to third party libraries, the general wisdom is "if it's not broken, don't fix it". Why? Because our understanding of the library code is probably incomplete. If the library doesn't have comprehensive test cases included, it's easy to break things up.

If we start going down this path in re-engineering syzygy code without having comprehensive test cases and a way to measure its performance, at some point bad things are going to happen without us noticing it...

I'm afraid of opening a can of worms really, by approving this one patch, making people to submit more of similar kind of patches..

Then of course the other option is to try to really take over the syzygy code... This should preferably start with choosing test positions (which represent all kind of situations rencountered - especially around 50 move rule).  Then creating a set of benchmark positions. Then adding syzygy support for fishtest. I know that personally I won't have capacity to lead this, but if somebody has time and resources, go ahead. It should probably start up as a separate branch, and then when things are mature enough it could be merged. This way also sanity testing needs to be done only once.

So to summarize:
- I'm against merging these kind of patches as stand-alone changes.
- But if somebody wants to make a big effort to really convert the whole syzygy into Stockfish code (including the necessary tests, and fishtest changes), then I would be supportive of such attempt...

**Comment by lucasart** [comment]
@zamar: I understand your pov. Without a proper way to test for correctness (including smp…) and performance, the risk/reward is not very appealing. I'll try to think of some test suite for syzygy, before continuing to work on this. Ideas welcome.

**Comment by lucasart** [comment]
@zamar. So, to clarify, I want to take over this code base, and gradually rewrite it in a clean, concise, and self documenting stype, on par with SF (or close enough). I will maintain a syzygy branch in my repo, and close this pull. First thing I will do is create some proper testing methodology for syzygy.

Eventually, when it's good enough, I will open a pull. Might take a while, though. Meanwhile, people interested are welcome to look at my syzygy branch, and test it. The more pair of eyes the better, as the syzygy code is particularly complicated.
